stan/math/prim/prob/wiener5_lpdf.hpp

```diff
@@ -0,0 +1,710 @@
+ * @param a The boundary separation
+ * @param w_value The relative starting point
+ * @param error The error tolerance
+ * @return 'n_terms_large_t' term
+ */
+template <bool Density, bool GradW>
+inline double wiener5_n_terms_largel_t(double y, double a, double w_value,
+                                       double error) noexcept {
+  const double two_error = 2.0 * error;
+  const double y_asq = y / square(a);
+  const double two_log_a = 2 * log(a);
+  const double log_y_asq = log(y) - two_log_a;
+  const double w = 1.0 - w_value;
+
+  const double n_1_factor = GradW ? 2 : 3;
+  static constexpr double PI_SQUARED = pi() * pi();
+
+  double n_1;
+  double n_2;
+  if (Density) {
+    n_1 = 1.0 / (pi() * sqrt(y_asq));
+    const double two_log_piy = -2.0 * (LOG_PI + log_y_asq + error);
+    n_2 = (two_log_piy >= 0) ? sqrt(two_log_piy / (PI_SQUARED * y_asq)) : 0.0;
+  } else {
+    n_1 = sqrt(n_1_factor / y_asq) / pi();
+    const double u_eps_arg
+        = GradW
+              ? log(4.0) - log(9.0) + 2.0 * LOG_PI + 3.0 * log_y_asq + two_error
+              : log(3.0) - log(5.0) + LOG_PI + 2.0 * log_y_asq + error;
+    const double u_eps = fmin(-1, u_eps_arg);
+    const double arg_mult = GradW ? 1 : (2.0 / PI_SQUARED / y_asq);
+    const double arg = -arg_mult * (u_eps - sqrt(-2.0 * u_eps - 2.0));
+    n_2 = GradW ? (arg > 0) ? sqrt(arg / y_asq) / pi() : n_1
+                : (arg > 0) ? sqrt(arg) : n_1;
+  }
+
+  return ceil(fmax(n_1, n_2));
+}
+
+/**
+ * Calculate the 'result' term and its sign for a wiener5 density or gradient
+ *
+ * @tparam Density Whether the calculation is for the density
+ * @tparam GradW Whether the calculation is for gradient w.r.t. 'w'
+ *
+ * @param y A scalar variable; the reaction time in seconds
+ * @param a The boundary separation
+ * @param w_value The relative starting point
+ * @param n_terms_small_t The n_terms_small_t term
+ * @param n_terms_large_t The n_terms_large_t term
+ * @return 'result' sum and its sign
+ */
+template <bool Density, bool GradW>
+inline std::tuple<double, int> wiener5_log_sum_exp(
+    double y, double a, double w_value, size_t n_terms_small_t,
+    size_t n_terms_large_t) noexcept {
+  const double y_asq = y / square(a);
+  const double w = 1.0 - w_value;
+  const bool small_n_terms_small_t
+      = Density ? (2 * n_terms_small_t <= n_terms_large_t)
+                : (2 * n_terms_small_t < n_terms_large_t);
+  const double scaling = small_n_terms_small_t ? inv(2.0 * y_asq) : y_asq / 2.0;
+
+  double prev_val = NEGATIVE_INFTY;
+  double current_val = NEGATIVE_INFTY;
+  int prev_sign = 1;
+  int current_sign = 1;
+
+  if (small_n_terms_small_t) {
+    const double mult = Density ? 1 : 3;
+    const double offset = GradW ? y_asq : 0;
+    const double sqrt_offset = sqrt(offset);
+    for (size_t k = n_terms_small_t; k >= 1; k--) {
+      const double wp2k = w + 2.0 * k;
+      const double wm2k = w - 2.0 * k;
+      int wp2k_sign = (wp2k > sqrt_offset) ?: 1 - 1;
```

Comment:
` ?: 1 - 1`, bug? Is this supposed to be `? 1 : -1`?

Reply:
Oh, true, that's a bug. I changed this. Tests still run successful.